media-sound/beets/beets-1.4.9-r3.ebuild

```diff
@@ -0,0 +1,184 @@
+		badfiles? (
+			media-libs/flac
+			media-sound/mp3val
+		)
+		chromaprint? (
+			dev-python/audioread[${PYTHON_MULTI_USEDEP}]
+			dev-python/pyacoustid[${PYTHON_MULTI_USEDEP}]
+			media-libs/chromaprint[tools]
+		)
+		discogs? ( dev-python/discogs-client[${PYTHON_MULTI_USEDEP}] )
+		ffmpeg? ( media-video/ffmpeg:0[encode] )
+		gstreamer? (
+			media-libs/gst-plugins-good:1.0
+			media-libs/gst-plugins-bad:1.0
+		)
+		icu? ( dev-db/sqlite[icu] )
+		lastfm? ( dev-python/pylast[${PYTHON_MULTI_USEDEP}] )
+		mpd? (
+			dev-python/bluelet[${PYTHON_MULTI_USEDEP}]
+			dev-python/python-mpd[${PYTHON_MULTI_USEDEP}]
+		)
+		replaygain? (
+			gstreamer? (
+				dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]
+				media-plugins/gst-plugins-libav:1.0
+			)
+			!gstreamer? ( media-sound/mp3gain )
+		)
+		thumbnail? (
+			dev-python/pyxdg[${PYTHON_MULTI_USEDEP}]
+			|| (
+				dev-python/pillow[${PYTHON_MULTI_USEDEP}]
+				media-gfx/imagemagick
+			)
+		)
+		webserver? (
+			dev-python/flask[${PYTHON_MULTI_USEDEP}]
+			cors? (
+				dev-python/flask-cors[${PYTHON_MULTI_USEDEP}]
+			)
+		)
+	')"
+BDEPEND="dev-python/sphinx"
+
+PATCHES=(
+	"${FILESDIR}/${PV}-0001-compatibility-with-breaking-changes-to-the-ast-modul.patch"
+)
+
+DOCS=( README.rst docs/changelog.rst )
+
+distutils_enable_tests pytest
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+
+	rm_use_plugins() {
+		[[ -n "${1}" ]] || die "rm_use_plugins: No use option given"
+		local use=${1}
+		local plugins=${use}
+		use ${use} && return
+		einfo "no ${use}:"
+		[[ $# -gt 1 ]] && plugins="${@:2}"
+		for arg in ${plugins[@]}; do
+			einfo "  removing ${arg}"
+			if [[ -e "beetsplug/${arg}.py" ]]; then
+				rm beetsplug/${arg}.py || die "Unable to remove ${arg} plugin"
+			fi
+			if [[ -d "beetsplug/${arg}" ]]; then
+				rm -r beetsplug/${arg} || die "Unable to remove ${arg} plugin"
+			fi
+			sed -e "s:'beetsplug.${arg}',::" -i setup.py || \
+				die "Unable to disable ${arg} plugin "
+		done
+	}
+
+	rm_use_plugins chromaprint chroma
+	rm_use_plugins ffmpeg convert
+	if ! use icu; then
+		rm_use_plugins icu loadext
+	fi
+	rm_use_plugins lastfm lastgenre lastimport
+	rm_use_plugins mpd bpd mpdstats
+	rm_use_plugins webserver web
+	rm_use_plugins thumbnail thumbnails
+
+	# remove plugins that do not have appropriate dependencies installed
+	for flag in badfiles discogs replaygain; do
+		rm_use_plugins ${flag}
+	done
+
+	if ! use mpd; then
+		rm test/test_player.py || die "Failed to remove test_player.py"
+		rm test/test_mpdstats.py || die "Failed to remove test_mpdstats.py"
+	fi
+	if ! use webserver; then
+		rm test/test_web.py || die "Failed to remove test_web.py"
+	fi
+	if ! use replaygain; then
+		rm test/test_replaygain.py || die "Failed to remove test_replaygain.py"
+	fi
+	if ! use ffmpeg; then
+		rm test/test_convert.py || die "Failed to remove test_convert.py"
+	fi
+	if ! use thumbnail; then
+		rm test/test_thumbnails.py || die "Failed to remove test_thumbnails.py"
+	fi
+	if use test; then
```

Comment:
Any reason not to remove them unconditionally?

Reply:
Because it can be usefull to test be it need the use flag enabled so I would prefer to keep them like that (more precise tests),
but please elaborate on this maybe I'm wrong (probably).

Reply:
I do not understand what you've said. I'm asking why do you need to `if use test` here if the files being removed unconditionally would not make any difference for when tests are disabled, and removing it would make the ebuild shorter and catch more mistakes with `USE=-test`.

Reply:
Thank you for the feedback,
I aggree the ebuild should be as simple as possible so I upgraded as you suggested.